modules/common/services/audio.nix

```diff
@@ -91,17 +91,70 @@ in
     # Start pipewire on system boot
     systemd.services.pipewire.wantedBy = [ "multi-user.target" ];
 
-    systemd.services."pulseaudio-set-profile" =
+    systemd.services."initialize-audio-profile" =
       let
-        pulse-set-profile = pkgs.writeShellScriptBin "pulseaudio-set-profile" ''
-          ${pkgs.pulseaudio}/bin/pactl set-card-profile "alsa_card.pci-0000_00_07.0-platform-skl_hda_dsp_generic" "HiFi (Headphones, Mic1, Mic2)"
-          ${pkgs.pulseaudio}/bin/pactl set-card-profile "alsa_card.pci-0000_00_07.0-platform-skl_hda_dsp_generic" "HiFi (Mic1, Mic2, Speaker)"
-        '';
+        initialize-audio-profile = pkgs.writeShellApplication {
+          name = "initialize-audio-profile";
+          runtimeInputs = [
+            pkgs.pulseaudio
+            pkgs.jq
+          ];
+          text = ''
+            function setProfile() {
+                if [[ $# == 2 ]] && [[ -n "$1" ]] && [[ -n "$2" ]]; then
+                    echo "Setting audio device profile: ($1 - $2)"
+                    pactl set-card-profile "$1" "$2"
+                fi
+            }
+
+            function findAudioProfiles() {
+              local audio_device_name active_profile profile_list
+                if [[ $# == 1 ]] && [[ -n "$1" ]]; then
```

Comment:
It should be enough to check that there's exactly one argument `if [[ $# -eq 1 ]]; then`